lib/logitech_receiver/hidpp20.py

```diff
@@ -357,53 +359,276 @@ def __len__(self):
 #
 
 
+class ReprogrammableKey(object):
+    """Information about a control present on a device with the `REPROG_CONTROLS` feature.
+    Ref: https://lekensteyn.nl/files/logitech/logitech_hidpp_2.0_specification_draft_2012-06-04.pdf
+    Read-only properties:
+    - index {int} -- index in the control ID table
+    - key {_NamedInt} -- the name of this control
+    - default_task {_NamedInt} -- the native function of this control
+    - flags {List[str]} -- flags set on the control
+    """
+    def __init__(self, device, index, cid, tid, flags):
+        self._device = device
+        self.index = index
+        self._cid = cid
+        self._tid = tid
+        self._flags = flags
+
+    @property
+    def key(self) -> _NamedInt:
+        return special_keys.CONTROL[self._cid]
+
+    @property
+    def default_task(self) -> _NamedInt:
+        # NOTE: This NamedInt is a bit mixed up, because its value is the Control ID
+        # while the name is the Control ID's native task. But this makes more sense
+        # than presenting details of controls vs tasks in the interface. The same
+        # convention applies to `mapped_to`, `remappable_to`, `remap` in `ReprogrammableKeyV4`.
+        task = str(special_keys.TASK[self._tid])
+        return _NamedInt(self._cid, task)
+
+    @property
+    def flags(self) -> List[str]:
+        return special_keys.KEY_FLAG.flag_names(self._flags)
+
+
+class ReprogrammableKeyV4(ReprogrammableKey):
+    """Information about a control present on a device with the `REPROG_CONTROLS_V4` feature.
+    Ref: https://lekensteyn.nl/files/logitech/x1b04_specialkeysmsebuttons.html
```

Comment:
Hi,
This looks like version 2 of the x1b04 specification. Version 4 is available at https://drive.google.com/file/d/10imcbmoxTJ1N510poGdsviEhoFfB_Ua4/view (according to changelog two additional capabilities are described)
(For more docs see https://github.com/Logitech/cpg-docs (work in progress) and https://drive.google.com/drive/u/0/folders/0BxbRzx7vEV7eWmgwazJ3NUFfQ28)

Reply:
Thanks! I will have a look, especially `forceRawXY` looks relevant.

Reply:
I think that all pointers to Logitech documents should be to the Google drive unless there is a version available at the Logitech Github repository.